lib/modules/manager/gitlabci/utils.ts

```diff
@@ -2,7 +2,7 @@ import { regEx } from '../../../util/regex';
 import { getDep } from '../dockerfile/extract';
 import type { PackageDependency } from '../types';
 
-const re = /!reference \[(.*?)\]/g;
+const re = /!reference \[(.*?)\]/gs;
```

Comment:
```suggestion
const re = /!reference \[[^\]]+\]/g;
```
i think this regex is better